src/crypto/common.rs

```diff
@@ -288,3 +300,35 @@ impl fmt::Display for FromDnskeyError {
 }
 
 impl error::Error for FromDnskeyError {}
+
+//----------- Tests ----------------------------------------------------------
+
+#[cfg(test)]
+mod tests {
+    use crate::crypto::common::rsa_encode;
+
+    #[test]
+    fn test_rsa_encode() {
+        assert_eq!(rsa_encode(&[], &[]), &[0]);
+
+        assert_eq!(rsa_encode(&[1], &[]), &[1, 1]);
+        assert_eq!(rsa_encode(&[], &[1]), &[0, 1]);
+        assert_eq!(rsa_encode(&[1], &[1]), &[1, 1, 1]);
+
+        assert_eq!(rsa_encode(&[0, 1], &[1]), &[1, 1, 1]);
+        assert_eq!(rsa_encode(&[1], &[0, 1]), &[1, 1, 1]);
+        assert_eq!(rsa_encode(&[0, 1], &[0, 1]), &[1, 1, 1]);
+
+        assert_eq!(rsa_encode(&[0, 0, 1], &[0, 1]), &[1, 1, 1]);
+        assert_eq!(rsa_encode(&[0, 1], &[0, 0, 1]), &[1, 1, 1]);
+        assert_eq!(rsa_encode(&[0, 1, 1], &[0, 0, 1]), &[2, 1, 1, 1]);
+
+        assert_eq!(rsa_encode(&[1, 2], &[]), &[2, 1, 2]);
+        assert_eq!(rsa_encode(&[], &[1, 2]), &[0, 1, 2]);
+        assert_eq!(rsa_encode(&[1, 2], &[1, 2]), &[2, 1, 2, 1, 2]);
+
+        assert_eq!(rsa_encode(&[0, 1, 2], &[1]), &[2, 1, 2, 1]);
+        assert_eq!(rsa_encode(&[1], &[0, 1, 2]), &[1, 1, 1, 2]);
+        assert_eq!(rsa_encode(&[0, 1, 2], &[0, 1, 2]), &[2, 1, 2, 1, 2]);
+    }
```

Comment:
Hmm, I didn't test the all zeroes case.

Reply:
I guess we can assume that keys are at least 512 bits big. If something weird happens if something is zero then that is fine.